lightningd/dual_open_control.c

```diff
@@ -1601,6 +1601,85 @@ static void handle_peer_tx_sigs_msg(struct subd *dualopend,
 				     channel->psbt);
 }
 
+static void handle_validate_rbf(struct subd *dualopend,
+				const u8 *msg)
+{
+	struct wally_psbt *candidate_psbt;
+	struct channel_inflight *inflight;
+	struct channel *channel = dualopend->channel;
+	struct wally_psbt *set_psbt;
+
+	if (!fromwire_dualopend_rbf_validate(tmpctx, msg,
+					     &candidate_psbt)) {
+		channel_internal_error(channel,
+				       "Malformed dualopend_rbf_validate %s",
+				       tal_hex(tmpctx, msg));
+		return;
+	}
+
+	/* Iterate through all previous inflights, confirm
```

Comment:
IMHO This logic is cleaner if we start with the new candidate, then iterate.  And we only need a flag per input, not an actual PSBT.

Approx:
```
bool *inputs_present = tal_arr(tmpctx, set_psbt->num_inputs);
memset(inputs_present, true, tal_bytelen(inputs_present));

list_for_each(&channel->inflights, inflight, list) {
    for (size_t i = 0; i < set_psbt->num_inputs; i++) {
       if (!psbt_has_input(inflight->funding_psbt, set_psbt->tx->inputs[i]))
            inputs_present[i] = false;
    }
}

if (memeqzero(inputs_present, tal_bytelen(inputs_present)) {
   error ...
```

Reply:
This is much nicer!